api/nodes/permissions.py

```diff
@@ -74,6 +75,32 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth)
             return has_auth
 
+class ContributorOrPublicForRelationshipPointers(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, dict)
+        auth = get_user_auth(request)
+        parent_node = obj['self']
+
+        if request.method in permissions.SAFE_METHODS:
+            return parent_node.can_view(auth)
+        elif request.method == 'DELETE':
+            return parent_node.can_edit(auth)
```

Comment:
The docstring says that this requires admin permissions, but you are only checking read+write here; can you please verify this?